NexusClient/Updating/ProgramUpdater.cs

```diff
@@ -0,0 +1,359 @@
+                        return CancelUpdate();
+                    }
+
+                    SetMessage("Launching installer...");
+                    var psiInfo = new ProcessStartInfo(newInstaller);
+                    Process.Start(psiInfo);
+                    Trace.Unindent();
+
+                    return true;
+                }
+            }
+            else if (!_isAutomaticCheck)
+            {
+                var promptMessage = 
+                    $"{CommonData.ModManagerName} is already up to date.{Environment.NewLine}{Environment.NewLine}" +
+                    $"NOTE: You can find the release notes and past versions here:{Environment.NewLine}{Links.Instance.Releases}";
+
+                DisplayDialog(() => ExtendedMessageBox.Show(null, promptMessage, "Up to date", MessageBoxButtons.OK, MessageBoxIcon.Information));
+            }
+
+            SetMessage($"{CommonData.ModManagerName} is already up to date.");
+            SetProgress(2);
+            Trace.Unindent();
+
+            return true;
+        }
+
+        private static void DisplayDialog(ThreadStart showMessage)
+        {
+            try
+            {
+                // The extended message box contains an ActiveX control which must be run in an STA thread,
+                // we can't control what thread this gets called on, so create one if we need to.
+
+                var apartmentState = ApartmentState.Unknown;
+                Thread.CurrentThread.TrySetApartmentState(apartmentState);
+                
+                if (apartmentState == ApartmentState.STA)
+                {
+                    showMessage();
+                }
+                else
+                {
+                    var messageThread = new Thread(showMessage);
+                    messageThread.SetApartmentState(ApartmentState.STA);
+                    messageThread.Start();
+                    messageThread.Join();
+                }
+            }
+            catch {}
+        }
+
+        /// <summary>
+        /// Cancels the update.
+        /// </summary>
+        /// <remarks>
+        /// This is a convenience method that allows the setting of the message and
+        /// the determination of the return value in one call.
+        /// </remarks>
+        /// <returns>Always <c>true</c>.</returns>
+        private bool CancelUpdate()
+        {
+            SetMessage($"Cancelled {CommonData.ModManagerName} update.");
+            SetProgress(2);
+
+            return true;
+        }
+
+        /// <summary>
+        /// Constructs a changelog for all releases between <see cref="currentVersion"/> and <see cref="newVersion"/>.
+        /// </summary>
+        /// <param name="currentVersion">The currently running version of NMM.</param>
+        /// <param name="newVersion">The new version of NMM available for download.</param>
+        /// <returns></returns>
+        private string ConstructChangeLog(Version currentVersion, Version newVersion)
```

Comment:
This is where the brunt of changes in this PR lies: parsing the release notes and (lightly) beautifying them.